hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/MetricsTest.java

```diff
@@ -205,6 +210,193 @@ public void usingContextAsync() {
         job.join();
     }
 
+    @Test
+    public void availableDuringJobExecution() {
```

Comment:
From the name of the test it seems like you would like to assert the metrics while the job is running. If that's so, this test is really undeterministic, you will probably catch the job running at most some of the time. I think you might want to use something like `TestProcessors.NoOutputSourceP` and its `proceedLatch` to make it deterministic.

Reply:
I think it is deterministic - `TestSources.itemStream` creates `StreamStage` which means that job will be still running during metrics are asserted. `filter` stage stop to emit items when it reaches `generatedItems` amount. It means that certain number can be checked in `assertTrueEventually` part. I believe this deterministic and sufficient, WDYT?